build-logic/src/main/kotlin/org/terasology/gradology/module_build.kt

```diff
@@ -59,27 +61,23 @@ class ModuleMetadataForGradle(private val moduleConfig: ModuleMetadata) {
      * @return a list of modules and whether each is optional
      */
     fun moduleDependencies(): List<Pair<GradleDependencyInfo, Boolean>> {
-        val gestaltDeps = moduleConfig.dependencies.filterNotNull().filterNot { it.id.toString() == "engine" }
+        val gestaltDeps = moduleConfig.dependencies.filterNotNull().filterNot { it.id.toString() == ENGINE_MODULE_NAME }
         return gestaltDeps.map { gradleModule(it) }
     }
 
-    private fun gradleModule(gestaltDependency: DependencyInfo): Pair<GradleDependencyInfo, Boolean> {
-        if (!gestaltDependency.minVersion.isSnapshot) {
-            // gestalt considers snapshots to satisfy a minimum requirement:
-            // https://github.com/MovingBlocks/gestalt/blob/fe1893821127/gestalt-module/src/main/java/org/terasology/naming/VersionRange.java#L58-L59
-            gestaltDependency.minVersion = gestaltDependency.minVersion.snapshot
-            // (maybe there's some way to do that with a custom gradle resolver?
-            // but making a resolver that only works that way on gestalt modules specifically
-            // sounds complicated.)
-        }
-
+    private fun versionStringFromGestaltDependency(gestaltDependency: DependencyInfo): String {
         val version = if (gestaltDependency.versionPredicate() is VersionRange) {
             gestaltDependency.versionPredicate().toString()
         } else {
             // TODO: gradle-compatible version expressions for gestalt dependencies
             //     https://github.com/MovingBlocks/gestalt/issues/114
-            "[${gestaltDependency.minVersion},)"                                                                                                                                                                   
+            "[${gestaltDependency.minVersion},)"                                       
         }
+        return version;
+    }
+
+    private fun gradleModule(gestaltDependency: DependencyInfo): Pair<GradleDependencyInfo, Boolean> {
+        val version = versionStringFromGestaltDependency(gestaltDependency)
```

Comment:
thought: noticing neither of these functions refer to fields of the class, i.e. in Java we would say "method could be static."

Kotlin doesn't have a `static` keyword for this, but classes do have what is essentially a static _companion_ object you can put things in for similar effect. (for example, see the `forProject` function in this class.)

Reply:
Uh, nice! I have to learn more Kotlin! 

Scala has the [same concept](https://docs.scala-lang.org/overviews/scala-book/companion-objects.html), shouldn't be that different I guess :upside_down_face: